### ISSUE TITLE
Regex cannot to be parsed

### ISSUE THREAD

**Issue by upeter** [limansky/mongoquery#1]
When using regular expressions in the mq interpolator a variable expected error is thrown. It seems that regex is not supported

```
import com.github.limansky.mongoquery.casbah._

scala> val term = "anything"
term: String = anything

scala> mq"""{naam: /^$term/i}"""
<console>:12: error: variable expected
       mq"""{naam: /^$term/i}"""

```

**Comment by limansky** [comment]
Good point. I'll check what I can do.

**Comment by limansky** [comment]
Sorry for long delay. As a workaround you can use it this way:

```
import com.github.limansky.mongoquery.casbah._
scala>  val term = "^(?i)anything".r
term: scala.util.matching.Regex = ^(?i)anything

scala> mq"""{naam: $term}"""
res3: com.mongodb.DBObject = { "naam" : { "$regex" : "^(?i)anything" , "$options" : "i"}}
```

Btw, for me, it's more clear usage. The reason is that we want to use regex and we passing regex. So, the regex is an atomic value, like string for example. And the behaviour is the same, you cannot combine string from the different parts like: `mq{ foo : "bar$x" }`. But, like for the other types it would be nice to support literals: `mq{ bar : /[0-9]+/ }`

So I'd like to use this issue to track regex literals support.

**Comment by limansky** [comment]
Btw, for me, it's more clear usage. The reason is that we want to use regex and we passing regex. So, the regex is an atomic value, like string for example. And the behaviour is the same, you cannot combine string from the different parts like: ```mq{ foo : "bar$x" }```. But, like for the other types it would be nice to support literals: ```mq{ bar : /[0-9]+/  }```

So I'd like to use this issue to track regex literals support.

**Comment by limansky** [comment]
Initial drop for regex literals support is available in 0.5-SNAPSHOT.
